schema/query/projection_parser.go

```diff
@@ -178,7 +178,7 @@ func (p *projectionParser) scanFieldName() string {
 	field := []byte{}
 	for p.more() {
 		c := p.peek()
-		if (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_' || c == '-' {
+		if (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_' || c == '-' || c == '*' {
```

Comment:
This will allow `*` in field names, which is not what we want right? The `*` should only be allowed by itself.

Reply:
Yes, will fix this.